pkgs/by-name/ms/msecli/package.nix

```diff
@@ -0,0 +1,47 @@
+{
+  autoPatchelfHook,
+  buildFHSEnv,
+  fetchurl,
+  lib,
+  stdenv,
+  zlib,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "msecli";
+  version = "10.01.012024.00";
+
+  src = fetchurl {
+    url = "https://www.micron.com/content/dam/micron/global/public/products/software/storage-executive-software/msecli/msecli-linux.run";
+    hash = "sha256-IszdD/9fAh+JA26bSR1roXSo8LDU/rf4CuRI3HjU1xc=";
+  };
+
+  buildInputs = [ zlib ];
+
+  nativeBuildInputs = [ autoPatchelfHook ];
+
+  unpackEnv = buildFHSEnv { name = "${pname}-unpackEnv"; };
+
+  unpackPhase = ''
+    cp "$src" ${src.name}
+    chmod +x ${src.name}
+    # ignore the exit code as the installer
+    # fails at optional steps due to read only FHS
+    ${unpackEnv}/bin/${unpackEnv.name} -c "./${src.name} --mode unattended --prefix bin || true"
```

Comment:
This should probably be called during the `buildPhase`?

Reply:
I can replace the unpackEnv with a buildEnv and shift the installer step into the buildPhase. Would that be sufficient?

Reply:
That sounds ideal, if it's not too much of a hassle.

Reply:
Not a hassle at all. Thank you for taking the time to review the PR.
Refactored and squashed.